pac4j-cas/src/main/java/org/pac4j/cas/client/CasClient.java

```diff
@@ -266,7 +266,7 @@ protected void initializeClientConfiguration() {
             this.casPrefixUrl += "/";
         }
         if (CommonHelper.isBlank(this.casPrefixUrl)) {
-            this.casPrefixUrl = this.casLoginUrl.replaceFirst("/login", "/");
+            this.casPrefixUrl = this.casLoginUrl.replaceFirst("/login$", "/$");
```

Comment:
The `$` is not a required convention, it's part of the regexp which is the first parameter of the `replaceFirst` method.

```
String url1 = "http://localhost:8080/cas/login";
System.out.println(url1.replaceFirst("/login$", "/end"));
```
Will output: `http://localhost:8080/cas/end`.

And it works for: `String url1 = "http://loginserver/cas/login";` which is the needed fix.

Though, the proposed fix: `this.casPrefixUrl = this.casLoginUrl.replaceFirst("/login$", "/$");` doesn't work and generates: `Illegal group reference: group index is missing`.